finality-aleph/src/party/mod.rs

```diff
@@ -432,9 +432,20 @@ where
     }
 
     pub async fn run(mut self) {
-        let last_finalized_number = self.client.info().finalized_number;
+        let mut finalized_number = self.client.info().finalized_number;
+        let mut previous_finalized_number;
+        loop {
```

Comment:
The above two lines are pretty unusual style for our code. I feel like it might be more readable if you:
1. explicitly initialize `previous_finalized_number`
2. replace the `loop` with a `while` – I think you could do 1. in a way that ensures the numbers differ (e.g. `finalized_number + 1`) and that would just work. also we save 1 second in the optimistic case :D
The second point would probably require a comment for why the initialization is weird, but this would still be more readable than this? Disputable I guess.